src/main/kotlin/calculator/NumberParser.kt

```diff
@@ -0,0 +1,21 @@
+package calculator
+
+class NumberParser {
+    private val delimiterStrategies = listOf(
+        CustomDelimiterStrategy(),
+        DefaultDelimiterStrategy()
+    )
+
+    fun parse(text: String): List<Int> {
+        return when {
+            text.length == 1 -> listOf(text.toInt())
+            else -> parseWithStrategy(text)
+        }
```

Comment:
text의 길이로만 case를 구분하실거라면 아래와 같이 변경할 수도 있을 것 같네요. 😃
```suggestion
        return when (text.length) {
            1 -> listOf(text.toInt())
            else -> parseWithStrategy(text)
        }
```

Reply:
정규식으로 판단 로직 변경하면서 해당 로직들은 없어졌지만, 추후 저런 케이스 있으면 저런식으로 쓰는게 가독성 더 좋아보이네요 ㅎㅎ 감사합니다~